cola/cmds.py

```diff
@@ -1319,11 +1319,34 @@ class LaunchEditor(Edit):
     def name():
         return N_('Launch Editor')
 
-    def __init__(self, context):
+    def __init__(self, context, widget=None):
```

Comment:
I imagine the following updates:

```diff
# in cola.widgets.diff
class DiffTextEdit(VimHintedPlainTextEdit):
    def _cursor_changed(self):
        """Update the line number display when the cursor changes"""
        line_number = max(0, self.textCursor().blockNumber())
        if self.numbers:
            self.numbers.set_highlighted(line_number)
+           newfile_line = self.numbers.newfile_line()


class DiffLineNumbers:
+    def newfile_line(self):
+        if self.lines and self.highlight_line >= 0:
+            # Find the next valid line
+            for i in range(self.highlight_line, len(self.lines)):
+                # take the "new" line number: last value in tuple
+                line_number = self.lines[i][-1]
+                if line_number > 0:
+                    return line_number
+
+            # Find the previous valid line
+            for i in range(self.highlight_line - 1, -1, -1):
+                # take the "new" line number: last value in tuple
+                line_number = self.lines[i][-1]
+                if line_number > 0:
+                    return line_number
```

But then, I don't quite see how the `context` could work to propagate that `newfile_line` value into the `context` of `LaunchEditor`.
I see the context gets created in `new_context`.
But after a quick look at the `SelectionModel`, I need your confirmation that the line_number fits (it's relevant to the current file, so maybe its place is near `SelectionModel.filename`. I guess the `line_number` does not have to survive a `union`).

Reply:
I was thinking we could just have a new member variable in the SelectionModel constructor that defaults to 0, e.g. `line_number`.  `self.context.selection.line_number = line_number` in the DiffEditor's callback can be used to store it.

We don't want that behavior on the base class because it's used by other widgets, which shouldn't be updating the semi-global line number.

When the diff command runs, it's the same context object, so you can get the value back out via the same self.context reference.